packages/medusa/src/commands/migrate.js

```diff
@@ -1,30 +1,43 @@
 import { createConnection } from "typeorm"
-import { getConfigFile } from "medusa-core-utils"
-
+import configLoader from "../loaders/config"
 import Logger from "../loaders/logger"
 
 import getMigrations from "./utils/get-migrations"
 
-const t = async function({ directory }) {
+const t = async function ({ directory }) {
   const args = process.argv
   args.shift()
   args.shift()
   args.shift()
+  const configModule = await configLoader(directory)
+  const migrationDirs = await getMigrations(directory)
+  let hostConfig = {
+    database: configModule.projectConfig.database_database,
+    url: configModule.projectConfig.database_url,
+  }
 
-  const { configModule } = getConfigFile(directory, `medusa-config`)
-  const migrationDirs = getMigrations(directory)
+  if (configModule.projectConfig.database_host) {
+    hostConfig = {
+      host: configModule.projectConfig.database_host,
+      port: configModule.projectConfig.database_port,
+      database: configModule.projectConfig.database_database,
+      ssl: configModule.projectConfig.database_ssl,
+      username: configModule.projectConfig.database_username,
+      password: configModule.projectConfig.database_password,
+    }
+  }
 
   const connection = await createConnection({
     type: configModule.projectConfig.database_type,
-    url: configModule.projectConfig.database_url,
-    extra: configModule.projectConfig.database_extra || {},
+    ...hostConfig,
+    extra: configModule?.projectConfig.database_extra || {},
     migrations: migrationDirs,
     logging: true,
   })
 
   if (args[0] === "run") {
-    await connection.runMigrations()
-    await connection.close()
+    await connection?.runMigrations()
```

Comment:
Question : why do we need that coalescing? The connection should always exists at this point. Same for next lines

Reply:
Oops,,, don't know how that crept in.. will clean it up